search-service/src/main/java/com/movie/recommender/search/controller/SearchController.java

```diff
@@ -0,0 +1,144 @@
+package com.movie.recommender.search.controller;
+
+import com.movie.recommender.common.model.movie.MovieDetails;
+import com.movie.recommender.search.exception.RatingError;
+import com.movie.recommender.search.service.SearchService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+@Slf4j
+@RestController
+@RequestMapping("api/v1/search")
+public class SearchController {
```

Comment:
there should be single search endpoint that takes a filter model (e.g., `SearchFilter`) as input
this model will encapsulate all the possible search criteria
`SearchService` can then handle the logic to process the filters and interact with the repository